examples/ensemble/plot_gradient_boosting_quantile.py

```diff
@@ -203,30 +203,52 @@ def highlight_min(x):
 # We can also evaluate the ability of the two extreme quantile estimators at
 # producing a well-calibrated conditional 90%-confidence interval.
 #
-# To do this we can compute the fraction of observations that fall between the
-# predictions:
+# To do this we can define two functions:
+#
+# - one to compute the fraction of observations that fall between the
+#   predictions;
+# - one to wrap those metrics in a cross-validation loop and assess their
+#   variability under data resampling.
+
+from sklearn.model_selection import KFold
+
+
 def coverage_fraction(y, y_low, y_high):
     return np.mean(np.logical_and(y >= y_low, y <= y_high))
 
 
-coverage_fraction(
-    y_train,
-    all_models["q 0.05"].predict(X_train),
-    all_models["q 0.95"].predict(X_train),
+def cross_val_coverage_fraction(models_dict, X, y, cv):
+    coverage_results = []
+    for _, test_index in cv.split(y):
+        y_low = models_dict["q 0.05"].predict(X[test_index])
+        y_high = models_dict["q 0.95"].predict(X[test_index])
+        y_true = y[test_index]
+
+        coverage = coverage_fraction(y_true, y_low, y_high)
+        coverage_results.append(coverage)
```

Comment:
It's more efficient to compute the predictions once, before resampling and then resample the predictions instead of the input features:

```suggestion
    predictions_low = models_dict["q 0.05"].predict(X)
    predictions_high = models_dict["q 0.95"].predict(X)
    for _, test_index in cv.split(y):
        y_low = predictions_low[test_index]
        y_high = predictions_high[test_index]
        y_true = y[test_index]

        coverage = coverage_fraction(y_true, y_low, y_high)
        coverage_results.append(coverage)
```

The same remark would apply for the boostrap method instead of using a CV splitter to resample.